master/buildbot/www/auth.py

```diff
@@ -111,13 +111,13 @@ def __init__(self, header=None, headerRegex=None, **kwargs):
         if self.userInfoProvider is None:
             self.userInfoProvider = UserInfoProviderBase()
         if header is not None:
-            self.header = header
+            self.header = unicode2bytes(header)
         if headerRegex is not None:
             self.headerRegex = re.compile(unicode2bytes(headerRegex))
 
     @defer.inlineCallbacks
     def maybeAutoLogin(self, request):
-        header = request.getHeader(self.header)
+        header = unicode2bytes(request.getHeader(self.header))
```

Comment:
request.getHeader() returns bytes, this change should be reverted

Reply:
I'm sorry but the backtrace in #4299 tells us otherwise:
```
 	File "/usr/lib/python3.6/site-packages/buildbot/www/auth.py", line 123, in maybeAutoLogin
 	    res = self.headerRegex.match(header)
 	builtins.TypeError: cannot use a bytes pattern on a string-like object
```

Reply:
Now, if `request.getHeader()` should absolutly not return a string (which it does according to this backtrace), maybe there is a bug elsewhere to discover.